### PR TITLE
Flesh out the card_declined error to include more info

### PR DESCRIPTION
**Pull request by embeaken**

### CODE REVIEW

File: lib/stripe_mock/api/errors.rb [hunk 1]
```diff
@@ -31,7 +31,14 @@ def self.argument_map
         invalid_cvc: ["The card's security code is invalid", 'cvc', 'invalid_cvc', 402],
         expired_card: ["The card has expired", 'exp_month', 'expired_card', 402],
         incorrect_cvc: ["The card's security code is incorrect", 'cvc', 'incorrect_cvc', 402],
-        card_declined: ["The card was declined", nil, 'card_declined', 402],
+        card_declined: [
+          "The card was declined",
+          nil,
+          'card_declined',
+          402,
+          { error: { message: "Your card was declined.", type: "card_error", code: "card_declined", decline_code: "generic_decline", charge: "ch_test" } }.to_json,
```
Comment (pjambet) [inline lib/stripe_mock/api/errors.rb]: why `to_json` on the first one only?

Comment (embeaken) [inline lib/stripe_mock/api/errors.rb]: That parameter is supposed to be the "HTTP body" which is the raw JSON string, and the next param is a Ruby hash in the actual Stripe response. So it does some JSON parsing.

Comment (pjambet) [inline lib/stripe_mock/api/errors.rb]: ok, I would just define the hash at the beginning of the method so that we don't recreated the same hash twice,

**Comment by pjambet** [comment]
:+1:

**Pull request merged**